src/app/services/views/ServiceDetailView.vue

```diff
@@ -18,120 +18,127 @@
         },
       ]"
     >
-      <div class="service-details">
-        <LoadingBlock v-if="isLoading" />
+      <DataSource
+        v-slot="{ data, isLoading, error }: ServiceInsightSource"
+        :src="`/${route.params.mesh}/service-insights/${route.params.service}`"
+      >
+        <DataSource
+          v-slot="{ data: externalService, isLoading: isLoadingExternalService, error: externalServiceError }: ExternalServiceSource"
+          :src="`/${route.params.mesh}/external-services/${route.params.service}`"
+        >
```

Comment:
Seems like this complexity should be pushed in the `sources` part of your datasource instead of here no?